test/ProductConstructionService.ScenarioTests/CodeFlowScenarioTestBase.cs

```diff
@@ -0,0 +1,185 @@
+        IReadOnlyList<Octokit.PullRequestFile> files = await GitHubApi.PullRequest.Files(TestParameters.GitHubTestOrg, targetRepoName, pullRequest.Number);
+
+        try
+        {
+            var versionDetailsFile = files.FirstOrDefault(file => file.FileName == "eng/Version.Details.xml");
+            versionDetailsFile.Should().NotBeNull();
+            versionDetailsFile!.Patch.Should().Contain(GetExpectedCodeFlowDependencyVersionEntry(sourceRepoName, commitSha));
+
+            // Verify new files are in the PR
+            foreach (var testFile in testFiles)
+            {
+                var newFile = files.FirstOrDefault(file => file.FileName == $"{testFile}");
+                newFile.Should().NotBeNull();
+                newFile!.Patch.Should().Be(testFilePatches[testFile]);
+            }
+        }
+        finally
+        {
+            // close the PR
+            await GitHubApi.PullRequest.Update(
+                TestParameters.GitHubTestOrg,
+                targetRepoName,
+                pullRequest.Number,
+                new Octokit.PullRequestUpdate
+                {
+                    State = Octokit.ItemState.Closed
+                });
+        }
+    }
+
+    protected async Task<AsyncDisposableValue<string>> CreateForwardFlowSubscriptionAsync(
+        string sourceChannelName,
+        string sourceRepo,
+        string targetRepo,
+        string targetBranch,
+        string updateFrequency,
+        string sourceOrg,
+        string targetDirectory)
+            => await CreateSourceEnabledSubscriptionAsync(
+                sourceChannelName,
+                sourceRepo,
+                targetRepo,
+                targetBranch,
+                updateFrequency,
+                sourceOrg,
+                targetDirectory: targetDirectory);
+
+    protected async Task<AsyncDisposableValue<string>> CreateBackwardFlowSubscriptionAsync(
+        string sourceChannelName,
+        string sourceRepo,
+        string targetRepo,
+        string targetBranch,
+        string updateFrequency,
+        string sourceOrg,
+        string sourceDirectory)
+            => await CreateSourceEnabledSubscriptionAsync(
+                sourceChannelName,
+                sourceRepo,
+                targetRepo,
+                targetBranch,
+                updateFrequency,
+                sourceOrg,
+                sourceDirectory: sourceDirectory);
+
+    protected async Task<AsyncDisposableValue<string>> CreateSourceEnabledSubscriptionAsync(
+        string sourceChannelName,
+        string sourceRepo,
+        string targetRepo,
+        string targetBranch,
+        string updateFrequency,
+        string sourceOrg = "dotnet",
+        bool sourceIsAzDo = false,
+        bool targetIsAzDo = false,
+        bool trigger = false,
+        string? sourceDirectory = null,
+        string? targetDirectory = null)
+    {
+        if (!string.IsNullOrEmpty(sourceDirectory) && !string.IsNullOrEmpty(targetDirectory))
```

Comment:
If we make this method protected, we could technically remove these since it will be a silent contract within this class and it would be upheld always

Reply:
Think you meant private here, so it's only accessible by the two other methods that call it. Done